File: 3797-design-spreadsheet/design-spreadsheet.cpp
class Spreadsheet {
    vector<vector<int>> sheet;
public:
    Spreadsheet(int rows) {
        sheet.assign(rows, vector<int>(26, 0));
    }
    
    void setCell(string cell, int value) {
        int c = cell[0] - 'A';
        int r = stoi(cell.substr(1)) - 1;
        sheet[r][c] = value;
    }
    
    void resetCell(string cell) {
        int c = cell[0] - 'A';
        int r = stoi(cell.substr(1)) - 1;
        sheet[r][c] = 0;
    }
    
    int getValue(string formula) {
        int p = -1;
        for (int i = 0, n = formula.size(); i < n; i++) {
            if (formula[i] == '+') {
                p = i;
                break;
            }
        }

        string x = formula.substr(1, p);
        string y = formula.substr(p + 1);
        int X = 0, Y = 0;
        if (isalpha(x[0])) {
            X = sheet[stoi(x.substr(1)) - 1][x[0] - 'A'];
        } else {
            X = stoi(x);
        }
        if (isalpha(y[0])) {
            Y = sheet[stoi(y.substr(1)) - 1][y[0] - 'A'];
        } else {
            Y = stoi(y);
        }

        return X + Y;
    }
};

/**
 * Your Spreadsheet object will be instantiated and called as such:
 * Spreadsheet* obj = new Spreadsheet(rows);
 * obj->setCell(cell,value);
 * obj->resetCell(cell);
 * int param_3 = obj->getValue(formula);
 */

File: 3797-design-spreadsheet/design-spreadsheet.java
class Spreadsheet {
    private int sheet[][];
    public Spreadsheet(int rows) {
        sheet = new int[rows][26];
        // Arrays.fill(sheet, 0);
    }
    
    public void setCell(String cell, int value) {
        int c = cell.charAt(0) - 'A';
        int r = Integer.parseInt(cell.substring(1)) - 1;
        sheet[r][c] = value;
    }
    
    public void resetCell(String cell) {
        int c = cell.charAt(0) - 'A';
        int r = Integer.parseInt(cell.substring(1)) - 1;
        sheet[r][c] = 0;
    }
    
    public int getValue(String formula) {
        int p = formula.indexOf('+');
        String x = formula.substring(1, p);
        String y = formula.substring(p + 1);

        int X = 0, Y = 0;
        if (x.charAt(0) >= 'A' && x.charAt(0) <= 'Z') {
            int c = x.charAt(0) - 'A';
            int r = Integer.parseInt(x.substring(1)) - 1;
            X = sheet[r][c];
        } else {
            X = Integer.parseInt(x);
        }
        if (y.charAt(0) >= 'A' && y.charAt(0) <= 'Z') {
            int c = y.charAt(0) - 'A';
            int r = Integer.parseInt(y.substring(1)) - 1;
            Y = sheet[r][c];
        } else {
            Y = Integer.parseInt(y);
        }

        return X + Y;
    }
}

/**
 * Your Spreadsheet object will be instantiated and called as such:
 * Spreadsheet obj = new Spreadsheet(rows);
 * obj.setCell(cell,value);
 * obj.resetCell(cell);
 * int param_3 = obj.getValue(formula);
 */